src/test/java/com/example/OnlineShopApiApplication/AdminServiceTests.java

```diff
@@ -0,0 +1,122 @@
+package com.example.OnlineShopApiApplication;
+
+import com.example.online_shop_api.Entity.Employee;
+import com.example.online_shop_api.Entity.Role;
+import com.example.online_shop_api.Repository.EmployeeRepository;
+import com.example.online_shop_api.Service.AdminService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class AdminServiceTests {
+    private AdminService adminService;
+    private EmployeeRepository employeeRepository;
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    private Employee testEmployee, testEmployee2;
+    private Role testRole;
+
+    @BeforeEach
+    public void setUp() {
+        employeeRepository = mock(EmployeeRepository.class);
+        bCryptPasswordEncoder = mock(BCryptPasswordEncoder.class);
+        adminService = new AdminService(employeeRepository);
+
+        testRole = new Role();
+        testRole.setId(2L);
+        testRole.setName("ROLE_EMPLOYEE");
+
+        testEmployee = Employee.builder()
+                .id(1L)
+                .firstName("Test1")
+                .lastName("Test1")
+                .username("TestUsername")
+                .email("test@mail.bg")
+                .password(bCryptPasswordEncoder.encode("TestPassword"))
+                .isEnabled(false)
+                .role(testRole)
+                .build();
+
+        testEmployee2 = Employee.builder()
```

Comment:
Seems that for most test methods you use just 1 employee. You can leave just 1 employee here in the @BeforeEach and add other employees only to methods that need them (the first test method).